client/state/action-types.js

```diff
@@ -401,6 +401,10 @@ export const PLUGIN_SETUP_INSTRUCTIONS_RECEIVE = 'PLUGIN_SETUP_INSTRUCTIONS_RECE
 export const PLUGIN_UPDATE_REQUEST = 'PLUGINS_UPDATE_REQUEST';
 export const PLUGIN_UPDATE_REQUEST_FAILURE = 'PLUGIN_UPDATE_REQUEST_FAILURE';
 export const PLUGIN_UPDATE_REQUEST_SUCCESS = 'PLUGIN_UPDATE_REQUEST_SUCCESS';
+export const PLUGIN_UPLOAD = 'PLUGIN_UPLOAD';
+export const PLUGIN_UPLOAD_UPDATE_PROGRESS = 'PLUGIN_UPLOAD_UPDATE_PROGRESS';
+export const PLUGIN_UPLOAD_COMPLETE = 'PLUGIN_UPLOAD_COMPLETE';
```

Comment:
There has been some discussion elsewhere around naming the actions relative to the _action_ rather than the api mechanics, meaning we move away from `SUCCESS` and `FAILURE`. My views on this are not strong, but yes, it is a shame that we have some actions with an easily replicated success/failure naming and others without.